agent-install/agent-uninstall.sh

```diff
@@ -0,0 +1,260 @@
+    -u          - exchange user authorization credentials
+    -d          - delete node from exchange
+
+Example: ./$(basename "$0") -u <hzn-exchange-user-auth> -d
+Note: Namespace may be stuck in "Terminating" during deleting. It is a known issue on kubernete. Please refer to Kubernete website to delete namespace mannually.
+
+EndOfMessage
+}
+
+function show_config() {
+	log_debug "show_config() begin"
+
+    echo "Current configuration:"
+    echo "HZN_EXCHANGE_USER_AUTH: ${HZN_EXCHANGE_USER_AUTH}"
+    echo "SKIP_UNREGISTRATION: ${SKIP_UNREGISTRATION}"
+    echo "Delete exchange node: ${DELETE_EX_NODE}"
+    echo "Verbosity is ${VERBOSITY}"
+
+    log_debug "show_config() end"
+}
+
+# checks input arguments and env variables specified
+function validate_args(){
+	log_debug "validate_args() begin"
+
+    log_info "Checking script requirements..."
+
+    # check kubectl is available
+	kubectl --help > /dev/null 2>&1
+	if [ $? -ne 0 ]; then
+            log_notify "kubectl is not available, please install kubectl and ensure that it is found on your \$PATH. Exiting..."
+            exit 1
+	fi
+
+    # check on mutual exclusive params (skip unregister=true and remove ex node=true)
+    if [[ "$SKIP_UNREGISTRATION" == "true" ]] && [[ "$DELETE_EX_NODE" == "true" ]]; then
+        log_notify "Cannot use '-d' if '-s' is used, exiting..."
+        exit 1
+    fi
+
+    # if need unregister node
+    if [[ "$SKIP_UNREGISTRATION" != "true" ]]; then
+        if [[ -z "$HZN_EXCHANGE_USER_AUTH" ]]; then
+            echo "\$HZN_EXCHANGE_USER_AUTH: ${HZN_EXCHANGE_USER_AUTH}_abc"
+            log_notify "\$HZN_EXCHANGE_USER_AUTH is not set. Exiting..."
+            exit 1
+        fi
+    fi
+    
+    log_info "Check finished successfully"
+    log_debug "validate_args() end"
+}
+
+function validate_number_int() {
+	log_debug "validate_number_int() begin"
+
+	re='^[0-9]+$'
+	if [[ $1 =~ $re ]] ; then
+   		# integer, validate if it's in a correct range
+   		if ! (($1 >= VERB_SILENT && $1 <= VERB_DEBUG)); then
+   			echo `now` "The verbosity number is not in range [${VERB_SILENT}; ${VERB_DEBUG}]."
+  			quit 2
+		fi
+   	else
+   		echo `now` "The provided verbosity value ${1} is not a number" >&2; quit 2
+	fi
+
+	log_debug "validate_number_int() end"
+}
+
+function get_agent_pod_id() {
+    log_debug "get_agent_pod_id() begin"
+    POD_ID=$(kubectl get pod -n ${NAMESPACE} 2> /dev/null | grep "agent-" | cut -d " " -f1 2> /dev/null)
+    if [ -n "${POD_ID}" ]; then
+        log_info "get pod: ${POD_ID}"
+    else
+        log_notify "Failed to get pod id, exiting..."
+        exit 1
+    fi
+    log_debug "get_agent_pod_id() end"
+}
+
+function unregister() {
+    log_debug "unregister() begin"
+    log_info "Unregister agent for pod: ${POD_ID}"
+
+    EXPORT_EX_USER_AUTH_CMD="export HZN_EXCHANGE_USER_AUTH=${HZN_EXCHANGE_USER_AUTH}"
+
+    if [[ "$DELETE_EX_NODE" == "true" ]]; then
+        log_info "This script will delete node from exchange"
```

Comment:
This script will delete the node from the management hub
similar comment for the else clause